woocommerce/woocommerce-gateway-stripe
# Hide WeChat Pay for ineligible Stripe account

Fixes #3145 

WeChat Pay shows up on the Stripe settings page even when the Stripe account country is not supported. In this PR weare hiding it from the settings page if the Stripe account country is not supported.

Find the supported countries and currencies [here](https://docs.stripe.com/payments/wechat-pay/accept-a-payment?ui=direct-api#supported-currencies)

## Changes proposed in this Pull Request:
- Using the `is_available_for_account_country` function to restrict WeChat Pay to the supported counties.
- Created `usePaymentMethodCurrencies`. This hook determines the supported currency for AliPay and WeChat Pay based on the Stripe account country. For other payment methods, it returns the fixed array of supported currencies.

## Testing instructions
- Connect to a US Stripe account and set the store currency USD. Confirm that you see WeChat Pay in the payment method list in `wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe&panel=methods`.
- Change the store currency to EURO. Confirm that you still see WeChat in the list but with the currency requirement pill.
- Change the store currency to CNY. Confirm that you still see WeChat in the list and there is no currency requirement pill.
- Connect to a Stripe account of an unsupported country (i.e, Mexico/Brazil/India). Confirm that WeChat Pay is not displayed in the settings page.
